aws/resource_aws_gamelift_queue.go

```diff
@@ -0,0 +1,200 @@
+							Optional:     true,
+							ValidateFunc: validation.IntAtLeast(0),
+						},
+					},
+				},
+			},
+			"timeout_in_seconds": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsGameliftQueueCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).gameliftconn
+
+	input := getFullInputCreate(d)
+	if v, ok := d.GetOk("name"); ok {
+		input.Name = aws.String(v.(string))
+	}
+	log.Printf("[INFO] Creating Gamelift Session Queue: %s", input)
+	out, err := conn.CreateGameSessionQueue(&input)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*out.GameSessionQueue.GameSessionQueueArn)
+	d.Set("name", out.GameSessionQueue.Name)
+
+	return resourceAwsGameliftQueueRead(d, meta)
+}
+
+func resourceAwsGameliftQueueRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).gameliftconn
+	log.Printf("[INFO] Describing Gamelift Session Queues: %s", d.Get("name"))
+	limit := int64(1)
+	out, err := conn.DescribeGameSessionQueues(&gamelift.DescribeGameSessionQueuesInput{
+		Names: aws.StringSlice([]string{d.Get("name").(string)}),
+		Limit: &limit,
+	})
+	if err != nil {
+		if isAWSErr(err, gamelift.ErrCodeNotFoundException, "") {
+			log.Printf("[WARN] Gamelift Session Queues (%s) not found, removing from state", d.Get("name"))
+			return nil
+		}
+		return err
+	}
+	sessionQueues := out.GameSessionQueues
+
+	if len(sessionQueues) < 1 {
+		log.Printf("[WARN] Gamelift Session Queue (%s) not found, removing from state", d.Get("name"))
+		return nil
+	}
+	if len(sessionQueues) != 1 {
+		return fmt.Errorf("expected exactly 1 Gamelift Session Queues, found %d under %q",
+			len(sessionQueues), d.Get("name"))
+	}
+	sessionQueue := sessionQueues[0]
+
+	d.Set("destinations", sessionQueue.Destinations)
+	d.Set("name", sessionQueue.Name)
+	d.Set("player_latency_policies", sessionQueue.PlayerLatencyPolicies)
+	d.Set("timeout_in_seconds", sessionQueue.TimeoutInSeconds)
+
+	return nil
+}
+
+func resourceAwsGameliftQueueUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).gameliftconn
+
+	name := d.Get("name").(string)
+
+	log.Printf("[INFO] Updating Gamelift Session Queue: %s", name)
+
+	if d.HasChange("name") || d.HasChange("destinations") ||
+		d.HasChange("player_latency_policies") || d.HasChange("timeout_in_seconds") {
+
+		input := getFullInputUpdate(d)
+
+		_, err := conn.UpdateGameSessionQueue(&input)
+		if err != nil {
+			return err
+		}
+	}
+
+	return resourceAwsGameliftQueueRead(d, meta)
+}
+
+func resourceAwsGameliftQueueDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).gameliftconn
+	name := d.Get("name").(string)
+	log.Printf("[INFO] Deleting Gamelift Session Queue: %s", name)
+	_, err := conn.DeleteGameSessionQueue(&gamelift.DeleteGameSessionQueueInput{
+		Name: aws.String(name),
+	})
+	if err != nil {
+		return err
```

Comment:
If the queue is previously deleted (e.g. outside Terraform), we should return successfully:

```go
if isAWSErr(err, gamelift.ErrCodeNotFoundException, "") {
  return nil
}
if err != nil {
  return fmt.Errorf("error deleting Gamelift Game Session Queue (%s): %s", d.Id(), err)
}
```